core/transaction.go

```diff
@@ -653,6 +668,21 @@ func transactionCommitmentPedersen(transactions []Transaction, protocolVersion s
 	return calculateCommitment(transactions, trie.RunOnTempTriePedersen, hashFunc)
 }
 
+// transactionCommitmentPoseidon0134 handles empty signatures compared to transactionCommitmentPoseidon:
+// empty signatures interpreted as [] instead of [0]
+func transactionCommitmentPoseidon0134(transactions []Transaction) (*felt.Felt, error) {
+	return calculateCommitment(transactions, trie.RunOnTempTriePoseidon, func(transaction Transaction) *felt.Felt {
+		var digest crypto.PoseidonDigest
+		digest.Update(transaction.Hash())
+
+		if txSignature := transaction.Signature(); len(txSignature) > 0 {
+			digest.Update(txSignature...)
+		}
+
+		return digest.Finish()
+	})
+}
+
 func transactionCommitmentPoseidon(transactions []Transaction) (*felt.Felt, error) {
```

Comment:
I would suggest to update the function name to include 0132 at the end for clarity

Reply:
I can add comment saying that Poseidon was used in < 0.13.2 version, wdyt?

Reply:
actually I thought about it one more time and decide to rename it :)